README.md

```diff
@@ -1,8 +1,102 @@
+    Note that this step might take around 45 minutes. <br />
+    Compiled Dataproc Driver JAR will be at `dataproc-jdbc-connector/jdbc-driver/build/libs/jdbc-driver-1.0-SNAPSHOT-all.jar`.
+
+2. Build example-client JAR
+    ```bash
+    ./gradlew -p example-client shadowJar
+    ```
+
+### Connection URL format
+* Dataproc Driver for Hive accepts JDBC URL string with prefix `jdbc:dataproc://hive/`
+* Client can pass in database name or leave it as empty to use default database <br>
+  `jdbc:dataproc://hive/;` or `jdbc:dataproc://hive/dbName;`
+* `projectId` and `region` are required parameters. To find your project and region, refer back to [Set up a Cloud project](https://cloud.google.com/dataproc/docs/guides/setup-project) in the **Before you begin** section above. <br>
+  `jdbc:dataproc://hive/;projectId={pid};region={region}`
+* `clusterName` is an optional parameter that allows client to specify the name of the cluster to connect to. <br>
+    `jdbc:dataproc://hive/;projectId={pid};region={region};clusterName={cluster-name}`
+* `clusterPoolLabel` is an optional parameter that supports submitting to a cluster pool. Client can pass in cluster pool labels and the Dataproc Driver will pick a healthy cluster (cluster with `status.state = ACTIVE`) within the pool to connect to. Please do not put `status.state` as one of the cluster pool label. 
+ 
+    Labels can be specified in the format `clusterPoolLabel=label1=value1[:label2=value2]`
+       
+    Example: 
+    ```bash
+    jdbc:dataproc://hive/;projectId={pid};region={region};clusteroPoolLabel=com=google:team=dataproc`
+    ```
+    Note that `clusterName` has a higher priority than `clusterPoolLabel`, if you pass in both parameters, the DataprocDriver will only look for the cluster by name.
+* DataprocDriver also accepts other semicolon separated list of session variables, Hive configuration variables or Hive variables that [Hive supports](https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-ConnectionURLFormat)
+
+    ```bash
+    jdbc:dataproc://hive/{dbName};projectId={pid};region={region};clusterName={name};sess_var_list?hive_conf_list#hive_var_list
+    ```
+    
+### Use with other JDBC client
+#### Using Beeline
+```bash
+# update the HADOOP_CLASSPATH to include the Dataproc JDBC Driver JARs
+export HADOOP_CLASSPATH=`hadoop classpath`:{path-to-driver-jar}jdbc-driver-1.0-SNAPSHOT-all.jar
+
+# tell beeline the class name for the driver using “-d” option
+beeline -d "com.google.cloud.dataproc.jdbc.DataprocDriver" -u "jdbc:dataproc://hive/;projectId={pid};region={region};{other-parameters}"
+```
+Example:
+```bash
+export HADOOP_CLASSPATH=`hadoop classpath`:/usr/local/home/Downloads/dataproc-jdbc-connector/jdbc-driver/build/libs/jdbc-driver-1.0-SNAPSHOT-all.jar
+beeline -d "com.google.cloud.dataproc.jdbc.DataprocDriver" -u "jdbc:dataproc://hive/;projectId=demo-dataproc;region=us-central1;clusterName=demo-cluster"
+```
+### Use with example client
```

Comment:
This section should talk about what the example client is -- an example usage of the JDBC driver from Java. It might even be worth pasting the important part of the client here, like:

```java
Connection connection = DriverManager.getConnection("jdbc:dataproc://hive/default;projectId=pId;region=us-central1;clusterName=my-cluster");
try (Statement stmt = connection.createStatement()) {
  ResultSet res = stmt.executeQuery("SHOW TABLES");
  while (res.hasNext()) {
    System.out.println(res.getString(1));
  }
}
```